src/avif.c

```diff
@@ -208,17 +208,41 @@ avifResult avifImageCopy(avifImage * dstImage, const avifImage * srcImage, avifP
             return allocationResult;
         }
     }
-    for (int plane = AVIF_CHAN_Y; plane <= AVIF_CHAN_A; ++plane) {
-        uint8_t * dstRow = avifImagePlane(dstImage, plane);
-        if (!dstRow) {
+    if (avifImageCopySamples(dstImage, srcImage, planes) != AVIF_RESULT_OK) {
+        assert(AVIF_FALSE);
+    }
+    return AVIF_RESULT_OK;
+}
+
+avifResult avifImageCopySamples(avifImage * dstImage, const avifImage * srcImage, avifPlanesFlags planes)
```

Comment:
This could go before it's first use in `avifImageCopy()`, but the ordering of functions is a bit unclear since this implements things from internal/avif.h and avif.h